### PR TITLE
Week1

### PR DESCRIPTION
**Pull request by jessiwitti**
R package with one initial function "compare_words".

### CODE REVIEW

File: GermanWordle/R/Week2functions.R [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#' @export
+target <- sample(german_words, 1)
+
+#' @title Compare Words
+#' @description This function compares whether the letters in the target word are the same as in the guessed word
+#' @param target_str character. Should be a word with the same length as the guessed word
+#' @param guess_str character. Should be a word with the same length as the target word
+#' 
+#' @return Shows whether the letter is in the word and on the correct place, in the word but incorrect place, or not in the word
```
Comment (brunykrijgsman) [inline GermanWordle/R/Week2functions.R]: Suggestion: Add the return type (array of strings?).

**Review by brunykrijgsman — APPROVED** [review]
- The R package skeleton looks good. Suggestion: You could add some things to your .gitignore (e.g., .DS-Store and .Rhistory).
- Clear function documentation (I made one suggestion in the code).

File: GermanWordle/R/Week2functions.R [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#' @export
+compare_words <- function(target_str, guess_str) {
+  if (nchar(target_str) != nchar(guess_str)) {
+    stop("target and guess string must be the same length.")
+  }
+  
+  target <- strsplit(target_str, "")[[1]]
+  guess <- strsplit(guess_str, "")[[1]]
+  result <- character(nchar(guess_str))
+
```
Comment (bdenizs) [inline GermanWordle/R/Week2functions.R:39]: Just a suggestion but you can add comments to your function and elaborate a bit more, and also check whether the word the users guessed is in the wordlist.

**Review by bdenizs — COMMENTED** [review]
In general I liked your wordle game. Your description was brief but explanatory as it's a very well-known game. Your R skeleton was complete and you did a good job on documentation, I just agree with the other review that you should add the return type. I also added a suggestion for the code :)

**Review by akane7011 — COMMENTED** [review]
The code looks concise and clear to me! 
Also depending on the rules, you may want to print the word length at the start of the game (or fixed to 5, for example).

File: GermanWordle/R/Week2functions.R (hunk 2 above)
Comment (akane7011) [inline GermanWordle/R/Week2functions.R:39]: It would be better to turn the input into lowercase letters - so that the capitalization doesn't matter here.

**Pull request merged**